pkg/controller/atlasdeployment/atlasdeployment_controller.go

```diff
@@ -310,11 +310,22 @@ func (r *AtlasDeploymentReconciler) handleDeletion(
 				return true, result
 			}
 		}
+		if err := r.cleanupBindings(context, deployment); err != nil {
```

Comment:
I think this cleanup should be done before removing the finalizer of the deployment itself. In the current way, if clean-up fails we won't be to try again because CR will be gone.

Reply:
makes sense, will move this call up

Reply:
fixed in next update